### PR TITLE
apache-flink-cdc 3.1.0 (new formula)

### PR DESCRIPTION
**Pull request by yuxiqian**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/a/apache-flink-cdc.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    url "https://search.maven.org/remotecontent?filepath=org/apache/flink/flink-cdc-pipeline-connector-values/3.1.0/flink-cdc-pipeline-connector-values-3.1.0.jar"
+    sha256 "7e16714c1395fa3139e60f189de54877020369e4ec2a7ec13cc4c7b84a26402e"
+  end
+
+  def install
+    libexec.install Dir["*"]
+    bin.write_exec_script libexec/"bin/flink-cdc.sh"
+    CONNECTORS.each do |connector|
+      (libexec/"lib/").install resource("#{connector}-connector")
+    end
```
Comment (SMillerDev) [inline Formula/a/apache-flink-cdc.rb:42]: You can just loop over the resources here.

File: Formula/a/apache-flink-cdc.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  end
+
+  test do
```
Comment (SMillerDev) [inline Formula/a/apache-flink-cdc.rb:45]: This should probably be configured to log to `var/"log"`

**Comment by yuxiqian** [comment]
Thanks for @SMillerDev's review! Addressed comments above.

File: Formula/a/apache-flink-cdc.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
+class ApacheFlinkCdc < Formula
+  desc "Flink CDC is a streaming data integration tool"
+  homepage "https://nightlies.apache.org/flink/flink-cdc-docs-stable"
+  url "https://www.apache.org/dyn/mirrors/mirrors.cgi?action=download&filename=flink/flink-cdc-3.1.0/flink-cdc-3.1.0-bin.tar.gz"
+  mirror "https://archive.apache.org/dist/flink/flink-cdc-3.1.0/flink-cdc-3.1.0-bin.tar.gz"
+  sha256 "f10b26a2368fd933c26b928fc118f1f1e28efec3ab58b0ec0c58f3f22906a9ab"
+  license "Apache-2.0"
+  head "https://github.com/apache/flink-cdc.git", branch: "master"
+  depends_on "apache-flink" => :test
```
Comment (SMillerDev) [inline Formula/a/apache-flink-cdc.rb:9]:
```suggestion
  head "https://github.com/apache/flink-cdc.git", branch: "master"

  depends_on "apache-flink" => :test
```

File: Formula/a/apache-flink-cdc.rb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+  def install
+    libexec.install Dir["*"]
+    bin.write_exec_script libexec/"bin/flink-cdc.sh"
+    resources.each do |connector|
+      (libexec/"lib/").install connector
+    end
+    (var/"log").mkpath
+    inreplace libexec/"bin/flink-cdc.sh" do |s|
+      s.sub! 'FLINK_CDC_LOG="$FLINK_CDC_HOME"/log', "FLINK_CDC_LOG=\"#{var/"log"}\""
```
Comment (SMillerDev) [inline Formula/a/apache-flink-cdc.rb:44]: Any other variables here that are not following normal filesystem conventions?

Comment (yuxiqian) [reply Formula/a/apache-flink-cdc.rb]: Thank you for pointing out this, moved log folder to `#{var}/log`, config files to `#{pkgetc}` and symlinked them to HOME path.

File: Formula/a/apache-flink-cdc.rb [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    libexec.install "lib"
+    resources.each { |connector| (libexec/"lib").install connector }
+
+    # Store configs in etc, outside of keg
+    pkgetc.install Dir["conf/*"]
+    libexec.install_symlink pkgetc => "conf"
+
+    # Create log folders
+    (var/"log/apache-flink-cdc").mkpath
+    libexec.install_symlink var/"log/apache-flink-cdc" => "log"
```
Comment (chenrui333) [inline Formula/a/apache-flink-cdc.rb:56]: this can be moved to post_install

File: Formula/a/apache-flink-cdc.rb [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  end
+  resource "values-connector" do
+    url "https://search.maven.org/remotecontent?filepath=org/apache/flink/flink-cdc-pipeline-connector-values/3.1.0/flink-cdc-pipeline-connector-values-3.1.0.jar"
+    sha256 "7e16714c1395fa3139e60f189de54877020369e4ec2a7ec13cc4c7b84a26402e"
+  end
+
+  def install
+    # Install launch script
+    libexec.install "bin"
+    bin.write_exec_script libexec/"bin/flink-cdc.sh"
```
Comment (chenrui333) [inline Formula/a/apache-flink-cdc.rb:40]:
```suggestion
    bin.write_exec_script libexec/"bin/flink-cdc.sh" => "flink-cdc"
```

File: Formula/a/apache-flink-cdc.rb [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+      sink:
+        name: Dummy data sink
+        type: values
+
+      pipeline:
+        name: Dummy pipeline job
+        parallelism: 1
+    EOS
+    (testpath/"log").mkpath
+    ENV.prepend "FLINK_LOG_DIR", testpath/"log"
```
Comment (chenrui333) [inline Formula/a/apache-flink-cdc.rb:74]: or it can be just `ENV["FLINK_LOG_DIR"] = testpath/"log"`

File: Formula/a/apache-flink-cdc.rb [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+    ENV.prepend "_JAVA_OPTIONS", "-Djava.io.tmpdir=#{testpath}"
```
Comment (chenrui333) [inline Formula/a/apache-flink-cdc.rb:75]: I think this is not needed?

File: Formula/a/apache-flink-cdc.rb (hunk 6 above)
Comment (yuxiqian) [reply Formula/a/apache-flink-cdc.rb]: Seems `write_exec_script` doesn't support creating alias. Added a `mv` to trim `.sh` extension.

Comment (chenrui333) [reply Formula/a/apache-flink-cdc.rb]: I see 👍

**Review by chenrui333 — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

**Pull request merged**